src/interlocked_hl.c

```diff
@@ -6,6 +6,8 @@
 
 #include "umock_c/umock_c_prod.h"
 
+#include "windows.h"
```

Comment:
>windows [](http://example.com/codeflow?start=10&length=7)

I don't think we should introduce windows-specific code in c-util, this won't work on Linux.

It looks like we first need a fix in c-pal so `wait_on_address` returns an enum and we can handle the various error cases there.

Reply:
Such a great comment!